### ISSUE TITLE
Access Denied error both opening & saving scripts

### ISSUE THREAD

**Issue by adrian-tt** [pgadmin-org/pgadmin4#7382]
** To Reproduce **

1. Open pgadmin4 in Firefox on default port 5050
2. Connect to postgreSQL v16 server (local) or v13 (remote)
3. Connect to database
4. Open query tool
5. Open File (both icon and Alt-Shift-O)
6. Select any *.sql file from the displayed list of files and folders (as expected) 
7. Click on 'Select' and query editor stays empty but Error dialog appear in bottom right corner with 'Access denied (/file-name-here.sql)
8. When trying to save the current the query, query response dialog changes from Data Output to Messages with 'Access denied (/save-file-name.sql)' message.

However I am able to create directories using the Save File dialog box but still unable to save a file in the new folder.

**Expected behavior**

1. Selecting a file listed in the Open File dialog should copy the contents into the query editor
2. Selecting or entering a new filename into the Save File dialog should save the contents of the query editor into the file specified

**Error message**

Access denied (/fred/s.sql)Access denied (/dd.sql)Access denied (/1)

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Debian, KDE Plasma
 - Version: 12, KDE 5.27.5
 - Mode: Desktop
 - Browser: Firefox v124.0.2 , Chrome v123.0.6312.122
 - Package type: Python wheel, pgadmin4-8.5-py3-none-any.whl

[pip3_modules.txt](https://github.com/pgadmin-org/pgadmin4/files/15038059/pip3_modules.txt)

Adding logging to the pgadmin config_local.py file generates the following log entries when openfile the File Open dialog and selecting an SQL script fromthe list of files

2024-04-19 09:55:01,966: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:01] "POST /file_manager/init HTTP/1.1" 200 -
2024-04-19 09:55:01,982: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:01] "POST /file_manager/filemanager/1285269/ HTTP/1.1" 200 -
2024-04-19 09:55:02,142: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:02] "POST /file_manager/save_last_dir/1285269 HTTP/1.1" 200 -
2024-04-19 09:55:02,495: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:02] "GET /sqleditor/status/383513 HTTP/1.1" 200 -
2024-04-19 09:55:04,006: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:04] "POST /file_manager/save_last_dir/1285269 HTTP/1.1" 200 -
2024-04-19 09:55:04,031: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:04] "DELETE /file_manager/delete_trans_id/1285269 HTTP/1.1" 200 -
**2024-04-19 09:55:04,035: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:04] "POST /sqleditor/load_file/ HTTP/1.1" 500 -**
2024-04-19 09:55:04,496: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:04] "GET /sqleditor/status/383513 HTTP/1.1" 200 -
2024-04-19 09:55:06,505: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:06] "GET /sqleditor/status/383513 HTTP/1.1" 200 -
2024-04-19 09:55:08,601: INFO   werkzeug:       127.0.0.1 - - [19/Apr/2024 09:55:08] "GET /sqleditor/status/383513 HTTP/1.1" 200 -

Previously worked with pgadmin4-8.4-py3-none-any.whl

Reinstalled v8.4 and same error so reinstalled v8.5. Access denied errors still occuring

**Comment by anilsahoo20** [comment]
Hi @adrian-tt 
We are not able to reproduce the issue. 
Can you please share the screen recording?

**Comment by adrian-tt** [comment]
Had already added rw to group and other to see if that worked (did not)

adrian@ah2022:~$ ls -la ~/.pgadmin/ 
total 6408
drwxrw-r--   5 adrian adrian    4096 Apr 19 14:46 .
drwxr-xr-x 191 adrian adrian   28672 Apr 19 12:41 ..
drwxrwxr-x   2 adrian adrian    4096 Jun 12  2023 azurecredentialcache
-rw-------   1 adrian adrian 4575232 Apr 19 14:46 pgadmin4.db
-rw-rw-r--   1 adrian adrian 1888256 Jan 16 17:57 pgadmin4.db.prev.bak
-rw-rw-r--   1 adrian adrian   42010 Apr 15 11:57 pgadmin4.log
drwxrw----   3 adrian adrian    4096 Apr 19 10:27 sessions
drwxrwxrwx   3 adrian adrian    4096 Jun 12  2023 storage


adrian@ah2022:~$ ls -la ~/.pgadmin/storage/ 
total 12
drwxrwxrwx  3 adrian adrian 4096 Jun 12  2023 .
drwxrw-r--  5 adrian adrian 4096 Apr 19 14:46 ..
drwxrwxrw- 22 adrian adrian 4096 Apr 16 08:50 adrian.heath_tracktrans.com


adrian@ah2022:~$ ls -la ~/.pgadmin/storage/adrian.heath_tracktrans.com/a*.sql
-rw-rwxrw- 1 adrian adrian    967 Apr 11  2019 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/add_ems_to_subs.sql
-rw-rwxrw- 1 adrian adrian    517 Apr  8  2020 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/aims_dab_log_query.sql
-rw-rwxrw- 1 adrian adrian   1494 May 22  2020 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/aims_feb_jobs.sql
-rw-rwxrw- 1 adrian adrian    378 Nov 15  2019 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/alto_loads.sql
-rw-rwxrw- 1 adrian adrian 401627 Jul 19  2021 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/aws_epodjobs_import.sql
-rw-rwxrw- 1 adrian adrian    864 Nov  1 15:44 /home/adrian/.pgadmin/storage/adrian.heath_tracktrans.com/aws_prod_emails.sql
[ah_pgadmin4-85.txt](https://github.com/pgadmin-org/pgadmin4/files/15041537/ah_pgadmin4-85.txt)

**Comment by adrian-tt** [comment]
Looks to be (data/file) corruption at my end.

**Steps to test & resolve**

1. Created new pgadmin user
2. Added one of the existing PostgreSQL servers
3. Connected to existing database
4. Opened query dialog, entered 'select current_database()' and successfully saved to test.sql file
5. Disconnected from psql server
6. Logged out of pgadmin
7. Stopped pgadmin
8. Renamed ~/.pgadmin/storage/original_user directory to original_user_broken
9. Copied ~/.pgadmin/storage/new_user to original_user
10. Restarted pgadmin
11. Logged in as original user
12. Connected to server and same database as (2)
13. Opened query dialog; newly created file from (4) visable in file list
14. Failed to open file
15. Logged out of pgadmin
16. Logged in as new_user
17. Deleted original user
18. Recreated original user
19. Copied storage/origin_user_broken files into storage/recreated_user
20. Logged in as recreated_user
21. Able to open original files and save new ones

**Comment by danidev05** [comment]
I was able to reproduce this issue after I saved in query tool 5 scripts with simple queries and used for a while with success. Then I performed a backup for another database and forgot to select a role name for this export operation. After this dump file was created I couldn't access anymore any file from the query tool: tried with query scripts created for the first database and I got same access denied for the exported file. Not only this, now I'm not able anymore to save a new script in query tool:
"Successfully run. Total query runtime: 160 msec.
3 rows affected.Access denied (/files-query)"

**Comment by AlexanderInova** [comment]
I run into the same issue (can list my files, but neither open or save files); only my user was affected. 

@anilsahoo20 Maybe this helps you to figure out what the cause if this issue is:
After checking the database `/var/lib/pgadmin/pgadmin4.db`, I recognized that my user had a preference `last_storage` set to `None`, while all other users had the preference set to `my_storage`. After changing the value for my user to `my_storage`, I was able to access open and write my files again.
